sdk/src/compressed_account.rs

```diff
@@ -262,67 +157,41 @@ where
 {
     input_account: T,
     output_account: T,
-    address_seed: Option<[u8; 32]>,
-    merkle_context: PackedMerkleContext,
-    merkle_tree_root_index: u16,
-    address_merkle_context: PackedAddressMerkleContext,
+    output_compressed_account: CompressedAccount,
 }
 
 impl<T> LightMutAccount<T>
 where
     T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Discriminator,
 {
-    pub fn try_from_slice(
-        v: &[u8],
-        merkle_context: &PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: &PackedAddressMerkleContext,
-    ) -> Result<Self> {
-        let account = T::try_from_slice(v)?;
+    pub fn try_from_slice(compressed_account: &CompressedAccount) -> Result<Self> {
+        let account = T::try_from_slice(&compressed_account.data.as_ref().unwrap().data)?;
+
+        // Clean the provided compressed account from data and use it as an
+        // output account. The data of output data is going to be assigned in
+        // the `output_compressed_account` method.
+        let output_compressed_account = CompressedAccount {
+            owner: compressed_account.owner,
```

Comment:
we should specify `owner` as a generic so that it doesn't occupy instruction data and is checked onchain.